src/main/java/org/folio/rest/impl/DocumentHelper.java

```diff
@@ -0,0 +1,66 @@
+package org.folio.rest.impl;
+
+import static org.folio.invoices.utils.HelperUtils.handleDeleteRequest;
+import static org.folio.invoices.utils.HelperUtils.handleGetRequest;
+import static org.folio.invoices.utils.ResourcePathResolver.*;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.invoices.utils.ResourcePathResolver;
+import org.folio.rest.jaxrs.model.DocumentCollection;
+import org.folio.rest.jaxrs.model.InvoiceDocument;
+
+import io.vertx.core.Context;
+import io.vertx.core.json.JsonObject;
+import me.escoffier.vertx.completablefuture.VertxCompletableFuture;
+
+class DocumentHelper extends AbstractHelper {
+
+  DocumentHelper(Map<String, String> okapiHeaders, Context ctx, String lang) {
+    super(getHttpClient(okapiHeaders), okapiHeaders, ctx, lang);
+  }
+
+  CompletableFuture<InvoiceDocument> createDocument(String invoiceId, InvoiceDocument document) {
+    JsonObject jsonDocument = JsonObject.mapFrom(document);
```

Comment:
According to jira desired approach is 
> If possible, reading the entire file's contents into memory should be avoided. A streaming approach would be ideal if feasible. 

As far as I can see RMB provides streaming capabilities only for [file uploads](https://github.com/folio-org/raml-module-builder/tree/master#file-uploads-option-2) where `application/octet-stream` is expected. So looks like there is no easy enough way to implement streaming approach...

Reply:
This is probably fine for the initial implementation.  If we find it's an issue we might consider changing the API to accept an application/octet-stream, e.g. separate endpoints for uploading metadata and file contents.